### ISSUE TITLE
When the connection fails and reconnects, you can't do anything, like the charge minion bug.

### ISSUE THREAD

**Issue by Joroc** [HearthSim/hs-bugs#438]

**Comment by Patashu** [comment]
More information needed -

What devices does it happen on? Does it happen all the time? If not, what causes it to happen?

Did it happen to you in Patch 4.2? If not, what is the most recent patch it occurred in?

**Comment by Joroc** [comment]
PC. All the time I dc. Most recent patch and the previous.

**Comment by culinko** [comment]
Can you explain the issue a bit more? What do you mean by "couldn't do anything"? Are you sure you reconnected on your turn? Was your opponent still playing cards? Did you see the rope? We need as many information as possible.

**Comment by Joroc** [comment]
It's still on my turn, there's the green outline around the minions and cards that can be played but i can only do one action then can't use the other cards and minions till restart of the client.
It will only accept the first attack or cast after the reconnection but the minion stops at the middle of the air, if i have other minions to attack, it's impossible.
Reconnecting restarting the client is fine.

**Comment by culinko** [comment]
How often does this happen to you? Do you remember any specific combination of cards that have been played by you or your opponent prior to the hang? This does not look like a connection issue, but an animation issue. Any chance that you play decks with return to hand effects or your opponent plays freezing trap (like mentioned in #363)?

**Comment by Joroc** [comment]
Yes, culinko

**Comment by PattuX** [comment]
Sounds like you just disconnect again.
The client will start the animation for you (lifting up the attacking minion) but fail to send information to the server (or recieve info frim it) because of the disconnect. It does not allow any further actions because the server didn't answer and there might be some information (secrets triggered for example) that would deny some actions. For the same reason the attacking animation does not continue but wait for information from the server that is never coming.

If this is right, the client will show you the disconnect message some seconds after you commended the attack. So does it pop up again? (might take half a minute or so sometimes)

Also, for how long is it fine after you restart client? Does it happen regularly or are there just certain times where it happens a lot and others where it never happens? Does it happen multiple times in a game?

**Comment by Joroc** [comment]
No. This first action is received by the server because when i restart it's done.
Happens everytime that my connection with the internet is failing and there's nothing with secret triggering or something like that.
Messages and rope will not appear.
If i restart it will reconect to the game and work as it should.

**Comment by Joroc** [comment]
A video of it https://youtu.be/8PpfLzCAsBQ

**Comment by culinko** [comment]
Thanks for the video! Could you enable logging and upload the Power.log the next time this happens? Here is how to enable logging (you just need the Power.log): https://github.com/jleclanche/fireplace/wiki/How-to-enable-logging. If you run Hearthstone Deck Tracker (or possibly other trackers), it's likely that you have the logging enabled already. Note: the Power.log is purged after restarting the client, so make sure to copy it right after it happens.

**Comment by Patashu** [comment]
http://us.battle.net/hearthstone/en/forum/topic/20743039747

Might be an occurrence of this?

**Comment by culinko** [comment]
Yes, it looks like this.

**Comment by Joroc** [comment]
[Power.txt](https://github.com/HearthSim/hs-bugs/files/217854/Power.txt)
Here it is. I guess.....

**Comment by Patashu** [comment]
http://us.battle.net/hearthstone/en/forum/topic/20743094612 might be an occurrence of this. If so, confirmed in 4.3.

**Comment by culinko** [comment]
Confirmed in 4.3.

https://www.reddit.com/r/hearthstone/comments/4fviyr/eu_disconnects/

**Comment by Patashu** [comment]
Possibly confirmed in 5.0?

http://us.battle.net/hearthstone/en/forum/topic/20743804520#new-post

**Comment by culinko** [comment]
Confirmed in 5.0 here: https://www.reddit.com/r/hearthstone/comments/4izwn6/reconnect_attempt_video/.

**Comment by Patashu** [comment]
http://us.battle.net/hearthstone/en/forum/topic/20745155259 "My game d/c'd for a few seconds and when I came back I couldn't click on my minions on the field. I was still able to use my hero power and play cards but that was about it."
